101-sql-vm-new-storage-ultrassd/azuredeploy.json

```diff
@@ -0,0 +1,408 @@
+      }
+    },
+    "dataPath": {
+      "type": "String",
+      "defaultValue": "F:\\SQLData",
+      "metadata": {
+        "description": "Path for SQL Data files"
+      }
+    },
+    "sqlLogUltraSSDDiskSizeInGB": {
+      "defaultValue": 512,
+      "type": "int",
+      "metadata": {
+        "description": "SQL Log UltraSSD Disk size in GiB."
+      }
+    },
+    "sqlLogUltraSSDdiskIOPSReadWrite": {
+      "defaultValue": 20000,
+      "type": "int",
+      "metadata": {
+        "description": "SQL Log UltraSSD Disk IOPS value representing the maximum IOPS that the disk can achieve."
+      }
+    },
+    "sqlLogUltraSSDdiskMbpsReadWrite": {
+      "defaultValue": 500,
+      "type": "int",
+      "metadata": {
+        "description": "SQL Log UltraSSD Disk MBps value representing the maximum throughput that the disk can achieve."
+      }
+    },
+    "logPath": {
+      "type": "String",
+      "defaultValue": "G:\\SQLLog",
+      "metadata": {
+        "description": "Path for SQL Log files"
+      }
+    },
+    "tempDbPath": {
+      "type": "String",
+      "defaultValue": "D:\\SQLTemp",
+      "metadata": {
+        "description": "Path for SQL Temp DB files"
+      }
+    },
+    "location": {
+      "type": "string",
+      "metadata": {
+        "description": "Location for all resources."
+      },
+      "defaultValue": "[resourceGroup().location]"
+    }
+  },
+  "variables": {
+    "networkInterfaceName": "[concat(parameters('virtualMachineName'), '-nic')]",
+    "networkSecurityGroupName": "[concat(parameters('virtualMachineName'), '-nsg')]",
+    "networkSecurityGroupRules": [
+      {
+        "name": "RDP",
+        "properties": {
+          "priority": 300,
+          "protocol": "TCP",
+          "access": "Allow",
+          "direction": "Inbound",
+          "sourceAddressPrefix": "*",
+          "sourcePortRange": "*",
+          "destinationAddressPrefix": "*",
+          "destinationPortRange": "3389"
+        }
+      }
+    ],
+    "publicIpAddressName": "[concat(parameters('virtualMachineName'), '-publicip-', uniqueString(parameters('virtualMachineName')))]",
+    "publicIpAddressType": "Dynamic",
+    "publicIpAddressSku": "Basic",
+    "diskConfigurationType": "NEW",
+    "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('networkSecurityGroupName'))]",
+    "subnetRef": "[resourceID(parameters('existingVNetResourceGroup'), 'Microsoft.Network/virtualNetWorks/subnets', parameters('existingVirtualNetworkName'), parameters('existingSubNetName'))]",
+    "dataDisksLuns": "[array(range(0 ,parameters('sqlDataDisksCount')))]",
+    "logDisksLuns": "[array(range(parameters('sqlDataDisksCount'), 1))]",
+    "dataDisks": {
+      "createOption": "empty",
+      "caching": "ReadOnly",
+      "writeAcceleratorEnabled": false,
+      "storageAccountType": "Premium_LRS",
+      "diskSizeGB": 1023
+    }
+  },
+  "resources": [
+    {
```

Comment:
let's remove the explicit disks and move the logic to the VM that needs the disks

Reply:
Ultra SSD needs to be created separately.

Reply:
Can you expand?  I don't see anything in the explicit declaration that can't be implicit off the top...  looks like it will all work fine.